src/core/Database.h

```diff
@@ -57,40 +60,33 @@ class Database : public QObject
     };
     static const quint32 CompressionAlgorithmMax = CompressionGZip;
 
-    struct DatabaseData
-    {
-        QUuid cipher;
-        CompressionAlgorithm compressionAlgo;
-        QByteArray transformedMasterKey;
-        QSharedPointer<Kdf> kdf;
-        QSharedPointer<const CompositeKey> key;
-        bool hasKey;
-        QByteArray masterSeed;
-        QByteArray challengeResponseKey;
-        QVariantMap publicCustomData;
-    };
-
     Database();
+    explicit Database(const QString& filePath);
     ~Database() override;
+
+    QUuid uuid() const;
+
+    bool open(QSharedPointer<const CompositeKey> key, bool readOnly = false, QString* error = nullptr);
+    bool open(const QString& filePath, QSharedPointer<const CompositeKey> key, bool readOnly = false, QString* error = nullptr);
+    bool save(bool atomic = true, bool backup = false, QString* error = nullptr);
+    bool save(const QString& filePath, bool atomic = true, bool backup = false, QString* error = nullptr);
+
+    QString filePath() const;
+    void setFilePath(const QString& filePath);
+    bool isReadOnly() const;
+    void setReadOnly(bool readOnly);
+    bool isInitialized() const;
+    void setInitialized(bool unlocked);
```

Comment:
There is a difference. Initialised means that I loaded stuff into the database. We need to make this distinction to find out if a database was populated with stuff (even if that stuff is "nothing") or if it's just an empty default instance. It doesn't have to be "unlocked" per-se. The Database object has no concept of being "unlocked". It only has contents or it doesn't, although no contents may be either an empty DB or it may be just a default instance which still needs to be populated.